### PR TITLE
feat: Try TypePHP

### PR DESCRIPTION
**Pull request by mitelg**
### 1. Why is this change necessary?

Try https://github.com/typephp-php/typephp

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [ ] I have written or adjusted the documentation and agent skills according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfilled them

### CODE REVIEW

File: src/Core/Framework/DependencyInjection/CompilerPass/McpToolAttributeReader.php [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 namespace Shopware\Core\Framework\DependencyInjection\CompilerPass;
 
+use Mcp\Capability\Attribute\McpTool;
 use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Mcp\Attribute\McpToolGroup;
```
Comment (mitelg) [inline src/Core/Framework/DependencyInjection/CompilerPass/McpToolAttributeReader.php]:
```suggestion
use Shopware\Core\Framework\Log\Package;
```

**Comment by rcalicdan** [comment]
@mitelg Hello, can bump the version typephp version to 0.3.4, I have fix some of the bugs discovered in the shopware CI

**Comment by rcalicdan** [comment]
Hello , I have review all the failing ci checks. All of them are now real type mismatch and docblock lies that Typephp has accurately detect, 

for example in this error:

```
  [TypePHP\Exception\TypeError]                                                
  Shopware\Core\Framework\DataAbstractionLayer\Doctrine\MultiInsertQueryQueue  
  ::prepareQueries(): Return value[0]['values'][1] must be of type string, in  
  t (1536232600) returned                                                      
                                                                               

Exception trace:
  at /home/runner/work/shopware/shopware/src/Core/Framework/DataAbstractionLayer/Doctrine/MultiInsertQueryQueue.php:12
 ```
 
 to fix this in n Shopware\Core\Framework\DataAbstractionLayer\Doctrine\MultiInsertQueryQueue.php, change line 97 from
 
 ```php
 /**
 * @return array<array{query: string, values: list<string>, types: list<ParameterType>}>
 */
 ```
 
 to 
 
 ```php
 /**
 * @return array<array{query: string, values: list<mixed>, types: list<ParameterType>}>
 */
 ```
 
 and btw, I have release version TypePHP v0.4.0. I had fixed many edge cases and added more features like late static binding detection..
 
 cheers,,
 
 Edit: I have fix and resolve many edge cases on the library you can update it to v0.5.1

**Comment by mitelg** [comment]
hey @rcalicdan 
I updated to latest version, but it seems, that there is now an issue again on your side :thinking: 
Also it feels that 0.4.x and 0.5.x even more, are slower than 0.3.x

**Comment by rcalicdan** [comment]
Hello @mitelg 

Based on the CI results, like the one shown here: https://github.com/shopware/shopware/actions/runs/32144816478/job/95736078382?pr=19140, I think the database connection was left in a hanging state after a sudden type error, or it may have slowed down due to other PR checks running concurrently. The error indicated that the issue was on TypePHP’s end. I’ve since fixed this bug.

Can you update the TypePHP's version to v0.5.4 and refresh the ci checks and see if the slow performance will persist,?

**Comment by rcalicdan** [comment]
Hello @mitelg , I notice in action: https://github.com/shopware/shopware/actions/runs/32234418245/job/96011239182?pr=19140 and its related check,  it seems that vendor isolation is not working properly and its executing vendor/src** in CI for migrations,,  I updated the librrary version to 0.5.5, and can you please update the CI checks? 

```bash
 PHP Fatal error:  Uncaught TypePHP\Exception\TypeError: Doctrine\DBAL\Schema\Column::__construct(): Argument $name must be of type Doctrine\DBAL\Schema\Name\UnqualifiedName, string '"id"' given in /home/runner/work/shopware/shopware/vendor/doctrine/dbal/src/Schema/AbstractNamedObject.php:31
```

**Comment by rcalicdan** [comment]
Hello @mitelg , I apologize for long reply and dupplicate PR.. I want to bring good news.. After sometime, I had finally able to optimize TypePHP performance to acceptable level and also resolve many edge cases. Take a look at this: https://github.com/rcalicdan/shopware/pull/1 , Performance has been massively improve

There's really a lot of docblock lies to fix in the shopware codebase. For integration test and php unit test that has a lot of type errors, It might be slow but this is due to Error Exception throwing but once that is resolve it would be faster. 

I recommend updating the typephp config in your branch to match this https://github.com/rcalicdan/shopware/blob/test-typephp/typephp.php to avoid type checking unnecessary files like migration folders and optimize array performance.

You can view the file changes in my fork https://github.com/rcalicdan/shopware/pull/1/changes and also you may notice that PHP stan now complain to a lot of things that a lot of runtime type errors has been resolve.

There's really a lot of type errors in the codebase, and maybe it's recommend to exclude more files for type checking.

**Comment by mitelg** [comment]
hey @rcalicdan 

thanks for your efforts you put into our code! I already thought that we will find much mis-leading PHP docs :sweat_smile: that is why I find your tool so interesting. unfortunately I do not have the time to fix all the stuff, but I will regularly try to do it step by step. nice to see, that we could bring your new tool to a good state with our cursed code base :sweat_smile:

**Comment by rcalicdan** [comment]
@mitelg  Thank you for the kind words..
